core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -207,8 +191,16 @@ public void runInContext() {
       });
       return;
     }
-    prepareHeaders(headers, callOptions, userAgent,
-        knownMessageEncodingRegistry, decompressorRegistry, compressorRegistry);
+    String compressorName = callOptions.getCompressor();
+    if (compressorName != null) {
+      compressor = compressorRegistry.lookupCompressor(compressorName);
+      checkArgument(compressor != null, "Unable to find compressor by name %s", compressorName);
+    }
+    if (compressor == null) {
```

Comment:
Should this be an else without any condition? Because of the checkArgument in the above compressor will only be null if compressorName is null.